server/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocateUnassignedDecision.java

```diff
@@ -258,33 +258,32 @@ public long getConfiguredDelayInMillis() {
     @Override
     public String getExplanation() {
         checkDecisionState();
-        AllocationDecision allocationDecision = getAllocationDecision();
-        if (allocationDecision == AllocationDecision.YES) {
-            return "can allocate the shard";
-        } else if (allocationDecision == AllocationDecision.THROTTLED) {
-            return "allocation temporarily throttled";
-        } else if (allocationDecision == AllocationDecision.AWAITING_INFO) {
-            return "cannot allocate because information about existing shard data is still being retrieved from some of the nodes";
-        } else if (allocationDecision == AllocationDecision.NO_VALID_SHARD_COPY) {
-            if (hasNodeWithStaleOrCorruptShard()) {
-                return "cannot allocate because all found copies of the shard are either stale or corrupt";
-            } else {
-                return "cannot allocate because a previous copy of the primary shard existed but can no longer be found on "
-                    + "the nodes in the cluster";
-            }
-        } else if (allocationDecision == AllocationDecision.ALLOCATION_DELAYED) {
-            return "cannot allocate because the cluster is still waiting "
+        return switch (getAllocationDecision()) {
+            case YES -> "Elasticsearch can allocate the shard.";
+            case THROTTLED -> "Elasticsearch is currently busy with other activities, but expects to be able to allocate this shard when "
+                + "those activities finish. Please wait.";
+            case AWAITING_INFO -> "Elasticsearch is currently retrieving information about this shard from one or more nodes, and will "
+                + "make an allocation decision once it receives the missing information. Please wait.";
+            case NO_VALID_SHARD_COPY -> (hasNodeWithStaleOrCorruptShard()
+                ? "Elasticsearch cannot allocate this shard because all the copies it found in the cluster are stale or corrupt. "
+                : "Elasticsearch cannot allocate this shard because it could not find any copies of its data in the cluster. ")
+                + "Elasticsearch will allocate this shard when a node holding a good copy of its data joins the cluster. "
+                + "If no such node is available then you should restore this index from a recent snapshot.";
```

Comment:
NIT:
```suggestion
            case NO_VALID_SHARD_COPY -> 
                "Elasticsearch cannot allocate this shard because "
                + (hasNodeWithStaleOrCorruptShard() ? "all the copies it found in the cluster are stale or corrupt. " : "it could not find any copies of its data in the cluster. ")
                + "Elasticsearch will allocate this shard when a node holding a good copy of its data joins the cluster. "
                + "If no such node is available then you should restore this index from a recent snapshot.";
```

Similarly to `NO` case tried to keep everything that is changing on the single line. It is slightly confusing when the line is starting with `:`.

Not sure if this suggestion actually makes it better